addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -373,7 +373,7 @@ def _timesheet_preprocess(self, vals_list):
                 vals['partner_id'] = data.partner_id.id
             if not vals.get('product_uom_id'):
                 company = account_per_id[vals['account_id']].company_id or data.company_id
-                vals['product_uom_id'] = uom_id_per_company.get(company.id, company.project_time_mode_id.id)
+                vals['product_uom_id'] = uom_id_per_company.get(company.id, company.project_time_mode_id.id) or self.env.company.project_time_mode_id.id
```

Comment:
I am not sure why we got 2 additional queries. 🤔 
1 query ok but 2 seems a bit strange. 🤔 

Can you check if you have time? 

Thanks in advance! 🙂